e2e/commands_test.go

```diff
@@ -124,179 +125,174 @@ maintainers:
+	icmd.RunCommand(dockerApp, "inspect").Assert(t, icmd.Success)
+	icmd.RunCommand(dockerApp, "inspect", ".").Assert(t, icmd.Success)
+	assert.NilError(t, os.Chdir(dir.Join("render")))
+	result := icmd.RunCommand(dockerApp, "inspect")
+	result.Assert(t, icmd.Expected{
+		ExitCode: 1,
+		Err:      "Error: multiple applications found in current directory, specify the application name on the command line",
+	})
 }
 
-func TestPackBinary(t *testing.T) {
-	dockerApp, hasExperimental := getDockerAppBinary(t)
-	if !hasExperimental {
-		t.Skip("experimental mode needed for this test")
-	}
+func TestPack(t *testing.T) {
+	skip.If(t, !hasExperimental, "experimental mode needed for this test")
 	tempDir, err := ioutil.TempDir("", "dockerapp")
 	assert.NilError(t, err)
 	defer os.RemoveAll(tempDir)
-	result := icmd.RunCommand(dockerApp, "pack", "helm", "-o", filepath.Join(tempDir, "test.dockerapp"))
-	result.Assert(t, icmd.Success)
+	icmd.RunCommand(dockerApp, "pack", "testdata/helm", "-o", filepath.Join(tempDir, "test.dockerapp")).Assert(t, icmd.Success)
 	// check that our commands run on the packed version
-	result = icmd.RunCommand(dockerApp, "inspect", filepath.Join(tempDir, "test"))
-	result.Assert(t, icmd.Success)
-	assert.Assert(t, strings.Contains(result.Stdout(), "myapp"), "got: %s", result.Stdout())
-	result = icmd.RunCommand(dockerApp, "render", filepath.Join(tempDir, "test"))
-	result.Assert(t, icmd.Success)
-	assert.Assert(t, strings.Contains(result.Stdout(), "nginx"))
+	icmd.RunCommand(dockerApp, "inspect", filepath.Join(tempDir, "test")).Assert(t, icmd.Expected{
+		Out: "myapp",
+	})
+	icmd.RunCommand(dockerApp, "render", filepath.Join(tempDir, "test")).Assert(t, icmd.Expected{
+		Out: "nginx",
+	})
 	cwd, err := os.Getwd()
 	assert.NilError(t, err)
 	assert.NilError(t, os.Chdir(tempDir))
 	defer func() { assert.NilError(t, os.Chdir(cwd)) }()
-	result = icmd.RunCommand(dockerApp, "helm", "test")
-	result.Assert(t, icmd.Success)
+	icmd.RunCommand(dockerApp, "helm", "test").Assert(t, icmd.Success)
 	_, err = os.Stat("test.chart/Chart.yaml")
 	assert.NilError(t, err)
 	assert.NilError(t, os.Mkdir("output", 0755))
-	result = icmd.RunCommand(dockerApp, "unpack", "test", "-o", "output")
-	result.Assert(t, icmd.Success)
+	icmd.RunCommand(dockerApp, "unpack", "test", "-o", "output").Assert(t, icmd.Success)
 	_, err = os.Stat("output/test.dockerapp/docker-compose.yml")
 	assert.NilError(t, err)
 }
 
-func runHelmCommand(t *testing.T, args ...string) *fs.Dir {
-	t.Helper()
-	dockerApp, _ := getDockerAppBinary(t)
-	abs, err := filepath.Abs(".")
-	assert.NilError(t, err)
-	dir := fs.NewDir(t, t.Name(), fs.FromDir(abs))
-	result := icmd.RunCmd(icmd.Cmd{
-		Command: append([]string{dockerApp}, args...),
-		Dir:     dir.Path(),
-	})
-	result.Assert(t, icmd.Success)
-	return dir
+func TestHelm(t *testing.T) {
+	t.Run("default", testHelm(""))
+	t.Run("v1beta1", testHelm("v1beta1"))
+	t.Run("v1beta2", testHelm("v1beta2"))
```

Comment:
This is here as, in the future `default` might not be `v1beta2` (and we should make sure `v1beta2`, in here, is a valid stack version)